src/plugins/orangehrmCorePlugin/Api/V2/Validator/Rules/IntArray.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Core\Api\V2\Validator\Rules;
+
+class IntArray extends AbstractRule
+{
+    /**
+     * @inheritDoc
+     */
+    public function validate($input): bool
+    {
+        if (!is_array($input)) {
+            return false;
+        }
+
+        // This will return an array with all $i elements that have type 'integer'
+        $filteredArray = array_filter($input, function ($i) {
+            return gettype($i) === 'integer';
```

Comment:
:thinking: 
```suggestion
            return is_int($i);
```

just another suggestion,
```php
    public function validate($input): bool
    {
        if (!is_array($input)) {
            return false;
        }

        foreach ($input as $item) {
            if (!is_int($item)) {
                return false;
            }
        }
        return true;
    }
```
Hope you have unit tests to evaluate this suggestion is correct

Reply:
changed to your suggested code. I had added tests in for KpiAPI